### ISSUE TITLE
bottleneck-1.0.0 conflict prevents package install and conda-build issue prevents attempted workaround

### ISSUE THREAD

**Issue by brantfaircloth** [conda/conda#2436]
Not sure what the issue is here, but it appears that since upgrading to anaconda 4.0.0/conda 4.0.5, I and other users of a package that I have built ([phyluce](https://github.com/faircloth-lab/conda-recipes/blob/master/recipes/phyluce/meta.yaml)) are getting the following dependency conflict when trying to install the `phyluce` package from my channel on anaconda.org:

```
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ......
Solving package specifications: ....

The following specifications were found to be in conflict:
  - bottleneck (target=bottleneck-1.0.0-np110py27_0.tar.bz2) -> numpy 1.10*|1.11*|1.9*
  - bottleneck (target=bottleneck-1.0.0-np110py27_0.tar.bz2) -> python 3.4*|3.5*
  - phyluce
Use "conda info <package>" to see the dependencies for each package.
```

The problem is that I don't entirely understand the nature of this error message.  I appears to suggest that `bottleneck-1.0.0` requires `numpy >=1.10`, but also suggests that `bottleneck`, built against `np110py27` has some dependency on `python 3.4 | 3.5`.  The conflict with my package (`phyluce`) may arise because the version of `biopython` (1.63) that `phyluce` currently requires is built against `numpy=1.7`, and it's possible that the numpy 1.7 version is causing the conflict with `bottleneck-1.0.0` - but the above error message is cryptic.

I've also tried to rebuild `phyluce`, requiring a newer version of `biopython` (1.66) that depends on a newer version of `numpy` (1.10), hoping to solve the conflict with `bottleneck`, but I'm running into a separate dependency resolution issue that appears to have been introduced to `conda-build` and which causes the build process to fail during a recursive build (may be similar to https://github.com/conda/conda-build/issues/849).

OS is OSX 10.10.5, conda info is below, and I am not sure why conda-build reports 0+unknown as the version in `conda info`:

```
$ conda info
Using Anaconda Cloud api site https://api.anaconda.org
Current conda install:

             platform : osx-64
        conda version : 4.0.5
  conda-build version : 0+unknown
       python version : 2.7.11.final.0
     requests version : 2.9.1
     root environment : /Users/bcf/anaconda-test  (writable)
  default environment : /Users/bcf/anaconda-test
     envs directories : /Users/bcf/anaconda-test/envs
        package cache : /Users/bcf/anaconda-test/pkgs
         channel URLs : https://conda.binstar.org/faircloth-lab/osx-64/
                        https://conda.binstar.org/faircloth-lab/noarch/
                        https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /Users/bcf/.condarc
    is foreign system : False
```

```
$ conda install conda-build
Using Anaconda Cloud api site https://api.anaconda.org
Fetching package metadata: ......
Solving package specifications: .........

# All requested packages already installed.
# packages in environment at /Users/bcf/anaconda-test:
#
conda-build               1.20.1                   py27_0
```

```
$ cat ~/.condarc
channels:
  - https://conda.binstar.org/faircloth-lab
  - defaults
```

**Comment by mcg1969** [comment]
It does look like `bottleneck` requires `numpy 1.9` or later so that is certainly enough to explain the conflict. Agreed that the hint is cryptic; we are working on that.

**Comment by brantfaircloth** [comment]
@mcg1969 thanks for the (super) quick response.  I'm going to try rebuilding with an earlier version of `conda-build` to escape the errors there and see if that solves the `bottleneck` problem.

**Comment by brantfaircloth** [comment]
No dice on that - with older `conda-build` (1.11.0), the build is failing because of conflicting package specifications:

```
Fetching package metadata: ........

WARNING: conda-build appears to be out of date. You have version 1.11.0 but the
latest version is 1.20.1. Run

conda update -n root conda-build

to get the latest version.

Solving package specifications: ....
Error: The following specifications were found to be in conflict:
  - phyluce 1.5.0 py27_1 -> biopython 1.66
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by mcg1969** [comment]
I'm not sure you're going to get past this without fixing the dependency issue. Taking a look at your `phyluce` packages, they have extremely rigid dependency specifications. For example, `phyluce-1.5.0-py27_1.tar.bz2` depends on `biopython 1.63`, so it makes sense that it would say that it's not compatible with `biopython 1.66`. 

That's just one example: every single dependency of phyluce is tied to an *exact* version. That's going to make it very easy to create version conflicts.

**Comment by brantfaircloth** [comment]
Ok.  Apologies in advance if this is uninformed/ignorant, but what I'm not understanding is that when I change the biopython dependency to 1.66 in my `meta.yaml`, then locally run the build on my machine to package up `phyluce`, it is still failing.

The rigid dependencies are meant to very specifically pin the versions of each build to their source releases so that users can know what version of programs they used to manipulate their data (the package is meant for scientific analysis - so this version tracking can be reasonably important). Most of the packages required by `phyluce` are also not python packages/libraries, and i use conda to setup a reasonable working environment that integrates these 3rd party binaries, python libraries, and the phyluce code to make things easier on the user.

**Comment by brantfaircloth** [comment]
I guess I should also clarify that I have built and packaged almost all of the many version-pinned dependencies (but not biopython) and make them available from the same channel where `phyluce` is found.

**Comment by lunchboxav** [comment]
interestingly, I also bumped into this exact same issue while trying to install `opencv`. @brantfaircloth, any luck on installing `phyluce` ?

**Comment by cfeng1** [comment]
I also have this problem when I try to install `graph-tool` from anaconda cloud. I'm fairly new to python, so I don't really know how to solve this. Any progress anyone??

**Comment by mcg1969** [comment]
I'm afraid it's difficult to diagnose these kinds of things---all the more reason why pinned specifications are problematic. Specifically, trying to mix pinned specifications like this with other packages that were not built with it from the outset. I understand the desire to construct reproducible environments but it can't be done piecemeal.

**Comment by msarahan** [comment]
Not sure it will help, but I've submitted a bottleneck recipe to
conda-forge: https://github.com/conda-forge/staged-recipes/pull/570.  I'm
not sure this will help, though - it looks like the defaults channel
already has a complete collection for numpy 1.10 and 1.11 - but not 1.9
(missing python 3.5).  If any part of your software is depending on numpy
1.9, it seems like this might be causing issues.

@lunchboxav and @cfeng1 - your issues might be related, but the exact
conflicts probably are not.  We won't be able to help you if you don't post
critical information including:

  - the output of ```conda list```
  - the package you're trying to install, including what channel you're
trying to install it from

**Comment by cfeng1** [comment]
Thank you @msarahan, I'm opening a new issue #2447.

**Comment by lunchboxav** [comment]
thank you for the suggestion @msarahan. I reference this issue in #2448 

Particularly, I'd like to know how can I change the numpy to version 1.9?

**Comment by kalefranz** [comment]
Should have been closed with #2447. Root causing being a package availability issue.
